core/operations/system.go

```diff
@@ -211,6 +211,19 @@ func (s *System) initializeVersionInfoHandler() {
 	s.mux.Handle("/version", s.handlerChain(versionInfo, false))
 }
 
+// RegisterHandler registers into the ServeMux a handler chain that borrows its security properties from the
+// operations.System. This method is thread safe because ServeMux.Handle() is thread safe, and options are immutable.
+// This method can be called either before or after System.Start(). If the pattern exists the method panics.
+func (s *System) RegisterHandler(pattern string, handler http.Handler) {
```

Comment:
so why is this a different pattern than the initialization of all the other stuff in the System? Such as metrics. Can't we adopt the previous pattern that is shown below?

```
func NewSystem(o Options) *System {
	logger := o.Logger
	logger := o.Logger
	if logger == nil {
	if logger == nil {
		logger = flogging.MustGetLogger("operations.runner")
		logger = flogging.MustGetLogger("operations.runner")
	}
	}
	system := &System{
	system := &System{
		logger:  logger,
		logger:  logger,
		options: o,
		options: o,
	}
	}
	system.initializeServer()
	system.initializeServer()
	system.initializeHealthCheckHandler()
	system.initializeHealthCheckHandler()
	system.initializeLoggingHandler()
	system.initializeLoggingHandler()
	system.initializeMetricsProvider()
	system.initializeMetricsProvider()
	system.initializeVersionInfoHandler()
	system.initializeVersionInfoHandler()
	return system
}
```
}